transport/transport.go

```diff
@@ -209,6 +223,8 @@ func (t *Transport) run(ctx context.Context) {
 		}
 	}()
 
+	t.table.AddPeer(t.self, wire.NewUnsignedAddress(wire.TCP, fmt.Sprintf("%v:%v", t.opts.Host, t.opts.Port), uint64(time.Now().UnixNano())))
```

Comment:
You need to add a test (or check to an existing test) that demonstrates that our own unsigned address is added/updated when the transport is run. This ensures future modifications do not break the assumption that this is being done.